### PR TITLE
chore: add the validation for Dynamic Link in listview

### PR DESCRIPTION
**Pull request by harshmule-epi**
Screenshot 2021-05-19 at 5 54 33 PM

Asana Task Link:
https://app.asana.com/0/1188592895233593/1199893804493442/f

### CODE REVIEW

File: frappe/public/js/frappe/form/controls/dynamic_link.js [hunk 1]
```diff
@@ -14,6 +14,10 @@ frappe.ui.form.ControlDynamicLink = frappe.ui.form.ControlLink.extend({
 			if (cur_list && cur_list.filter_area) {
 				// for list page
 				input = cur_list.filter_area.standard_filters_wrapper.find(selector);
+				if (!input.val()){
+					var msg = `Please Select ${this.df.options}`
```
Comment (sahil28297) [inline frappe/public/js/frappe/form/controls/dynamic_link.js]: You are currently displaying the fieldname, find a way to display the label of the field.

**Review by sahil28297 — CHANGES_REQUESTED** [review]

File: frappe/public/js/frappe/form/controls/dynamic_link.js (hunk 1 above)
Comment (harshmule-epi) [reply frappe/public/js/frappe/form/controls/dynamic_link.js]: okay!

Comment (harshmule-epi) [reply frappe/public/js/frappe/form/controls/dynamic_link.js]: Screenshot 2021-05-20 at 6 09 49 PM

File: frappe/public/js/frappe/form/controls/dynamic_link.js [hunk 2]
```diff
@@ -14,6 +14,10 @@ frappe.ui.form.ControlDynamicLink = frappe.ui.form.ControlLink.extend({
 			if (cur_list && cur_list.filter_area) {
 				// for list page
 				input = cur_list.filter_area.standard_filters_wrapper.find(selector);
+				if (!input.val()){
+					var msg = `Please select '${input[0]['placeholder']}' before selecting ${this.input.placeholder}`
```
Comment (vishdha) [inline frappe/public/js/frappe/form/controls/dynamic_link.js:18]:
```suggestion
					let msg = `Please select '${input[0]['placeholder']}' before selecting ${this.input.placeholder}`
```

**Review by vishdha — CHANGES_REQUESTED** [review]

File: frappe/public/js/frappe/form/controls/dynamic_link.js (hunk 2 above)
Comment (vishdha) [inline frappe/public/js/frappe/form/controls/dynamic_link.js]:
```suggestion
					frappe.msgprint(__(`Please select '${input[0]['placeholder']}' before selecting ${this.input.placeholder}`));
```

**Review by vishdha — CHANGES_REQUESTED** [review]

File: frappe/public/js/frappe/form/controls/dynamic_link.js [hunk 3]
```diff
@@ -14,6 +14,10 @@ frappe.ui.form.ControlDynamicLink = frappe.ui.form.ControlLink.extend({
...
+					frappe.msgprint(msg)
```
Comment (vishdha) [inline frappe/public/js/frappe/form/controls/dynamic_link.js]:
```suggestion
```

**Review by vishdha — CHANGES_REQUESTED** [review]

File: frappe/public/js/frappe/form/controls/dynamic_link.js [hunk 4]
```diff
@@ -14,6 +14,9 @@ frappe.ui.form.ControlDynamicLink = frappe.ui.form.ControlLink.extend({
 			if (cur_list && cur_list.filter_area) {
 				// for list page
 				input = cur_list.filter_area.standard_filters_wrapper.find(selector);
+				if (!input.val()){
+					frappe.msgprint(__(`Please select '${input[0]['placeholder']}' before selecting ${this.input.placeholder}`));
```
Comment (hrwX) [inline frappe/public/js/frappe/form/controls/dynamic_link.js]: - Not the placeholder, you have to show the docfield's label

**Review by hrwX — CHANGES_REQUESTED** [review]

File: frappe/public/js/frappe/form/controls/dynamic_link.js (hunk 4 above)
Comment (harshmule-epi) [reply frappe/public/js/frappe/form/controls/dynamic_link.js]: sure, will try to get label!

File: frappe/public/js/frappe/form/controls/dynamic_link.js [hunk 5]
```diff
@@ -14,6 +14,10 @@ frappe.ui.form.ControlDynamicLink = frappe.ui.form.ControlLink.extend({
 			if (cur_list && cur_list.filter_area) {
 				// for list page
 				input = cur_list.filter_area.standard_filters_wrapper.find(selector);
+				if (!input.val()){
+					let label = frappe.meta.get_label(this.df.doctype, this.df.options)
+					frappe.msgprint(__(`Please select '${label}' before selecting ${this.df.label}`));
```
Comment (hrwX) [inline frappe/public/js/frappe/form/controls/dynamic_link.js]:
```suggestion
					frappe.msgprint(__(`Please select <b>${label}</b> before selecting <b>${this.df.label}</b>`));
```

**Review by hrwX — DISMISSED** [review]

**Review by sahil28297 — APPROVED** [review]

**Review by hrwX — DISMISSED** [review]

File: frappe/public/js/frappe/form/controls/dynamic_link.js [hunk 6]
```diff
@@ -14,6 +14,10 @@ frappe.ui.form.ControlDynamicLink = frappe.ui.form.ControlLink.extend({
 			if (cur_list && cur_list.filter_area) {
 				// for list page
 				input = cur_list.filter_area.standard_filters_wrapper.find(selector);
+				if (!input.val()){
+					let label = frappe.meta.get_label(this.df.doctype, this.df.options)
```
Comment (sahil28297) [inline frappe/public/js/frappe/form/controls/dynamic_link.js]:
```suggestion
				if (!input.val()) {
					let label = frappe.meta.get_label(this.df.doctype, this.df.options);
```

**Review by sahil28297 — CHANGES_REQUESTED** [review]

**Review by sahil28297 — APPROVED** [review]

**Pull request merged**